src/main/java/org/freedesktop/secret/simple/SimpleCollection.java

```diff
@@ -206,6 +205,36 @@ public static boolean isAvailable() {
         }
     }
 
+    /**
+     * Checks if private/unsupported services are provided by the system:<br>
+     * <code>org.gnome.keyring</code>
+     *
+     * @return true if the secret service provider is gnome keyring, otherwise false and will log a warning message.
+     */
+    public static boolean isGnomeKeyringAvailable() {
+        if (connection != null && connection.isConnected()) {
+            try {
+                DBus bus = connection.getRemoteObject(
+                        Static.DBus.Service.DBUS,
+                        Static.DBus.ObjectPaths.DBUS,
+                        DBus.class);
+                List<String> names = Arrays.asList(bus.ListNames());
+                if (!(names.contains(
+                        org.gnome.keyring.Static.Service.KEYRING))) {
```

Comment:
Maybe it would be better to check if the actual `InternalUnsupportedGuiltRiddenInterface` is exposed and not only if the `gnome-keyring` is present. As the `gnome-keyring-daemon` also starts the `ssh-agent`. But this is somewhat of an edge case, which in practice should not be relevant, as one either installs the `gnome-keyring` and uses it mostly with default config or does not use it at all.

Reply:
```
/usr/bin/gnome-keyring-daemon --daemonize --login

/usr/bin/gnome-keyring-daemon --start --components=ssh
  /usr/bin/ssh-agent -D -a /run/user/1000/keyring/.ssh

/usr/bin/gnome-keyring-daemon --start --foreground --component=secrets
```

Reply:
The only method failing, when the `gnome-keyring-daemon` is not running is `isLocked()`. Invoking the [test for `isLocked()`](https://github.com/swiesend/secret-service/blob/46a49680fdabfa1998ad9b76a5dafca43f2d2e4b/src/test/java/org/freedesktop/secret/simple/SimpleCollectionTest.java#L255) actually activates the service again.

Reply:
The [RegressionTest](https://github.com/swiesend/secret-service/blob/46a49680fdabfa1998ad9b76a5dafca43f2d2e4b/src/test/java/org/freedesktop/secret/integration/RegressionTests.java#L26) failed for the same reason. They expected the collection to be unlocked in order to test unrelated things.